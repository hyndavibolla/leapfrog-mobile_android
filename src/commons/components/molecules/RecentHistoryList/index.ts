export { default as RecentHistoryList } from './RecentHistoryList';
