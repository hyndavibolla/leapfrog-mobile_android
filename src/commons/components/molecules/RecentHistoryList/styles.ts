import { css } from '@emotion/native';

import { COLOR, FONT_FAMILY, FONT_SIZE } from '_constants';

export const styles = {
  containerSearchHistory: css`
    margin-top: 16px;
  `,
  titleSearchHistory: css`
    font-family: ${FONT_FAMILY.BOLD};
    font-size: ${FONT_SIZE.SMALLER};
    line-height: ${FONT_SIZE.BIG};
    color: ${COLOR.BLACK};
    margin-bottom: 4px;
  `
};
