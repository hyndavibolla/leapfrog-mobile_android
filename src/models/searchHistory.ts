export enum RecentSearchHistoryType {
  REWARD = 'reward',
  MISSION = 'mission',
  CATEGORY = 'category '
}

export interface IRecentSearchHistory {
  id: string;
  name: string;
  type: RecentSearchHistoryType;
}
