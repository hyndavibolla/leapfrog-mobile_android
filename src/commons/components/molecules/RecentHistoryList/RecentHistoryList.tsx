import React, { memo } from 'react';
import { Text, View, ViewStyle, StyleProp } from 'react-native';

import { RecentHistoryItem } from '../RecentHistoryItem';
import { IRecentSearchHistory, RecentSearchHistoryType } from '_models/searchHistory';
import { useTestingHelper } from '_utils/useTestingHelper';

import { styles } from './styles';

export interface Props {
  title?: string;
  style?: StyleProp<ViewStyle>;
  visible?: boolean;
  historyList: IRecentSearchHistory[];
  onPress: (id: string | number, name: string, type: string) => void;
}

export const RecentHistoryList = ({ title, style, historyList, onPress, visible = true }: Props) => {
  const { getTestIdProps } = useTestingHelper('recent-search');

  return !historyList?.length || !visible ? null : (
    <View {...getTestIdProps('list')} style={[styles.containerSearchHistory, style]}>
      <Text style={styles.titleSearchHistory}>{title || 'Recent Search History'}</Text>
      {historyList.map(({ id, name, type }: IRecentSearchHistory) => {
        return <RecentHistoryItem key={id} onPress={() => onPress(id, name, type)} title={name} category={type === RecentSearchHistoryType.CATEGORY} />;
      })}
    </View>
  );
};

export default memo(RecentHistoryList);
