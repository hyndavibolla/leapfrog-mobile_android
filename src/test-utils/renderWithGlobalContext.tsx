import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
