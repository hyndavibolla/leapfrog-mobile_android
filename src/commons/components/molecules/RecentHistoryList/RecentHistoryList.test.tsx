import React from 'react';
import { fireEvent } from '@testing-library/react-native';

import RecentHistoryList, { Props } from './RecentHistoryList';
import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import { RecentSearchHistoryType, IRecentSearchHistory } from '_models/searchHistory';

describe('RecentHistoryList', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      onPress: jest.fn(),
      historyList: []
    };
  });

  it('should render', () => {
    const { toJSON } = renderWithGlobalContext(<RecentHistoryList {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with identifier', () => {
    const { queryByTestId } = renderWithGlobalContext(<RecentHistoryList {...props} />);
    expect(queryByTestId('recent-search-list')).toBeFalsy();
  });

  it('should execute onPress', async () => {
    const item: IRecentSearchHistory = { id: 'id', name: 'name', type: RecentSearchHistoryType.MISSION };
    props.historyList = [item];
    const { getByTestId } = renderWithGlobalContext(<RecentHistoryList {...props} />);
    fireEvent.press(getByTestId('recent-search-item'));
    expect(props.onPress).toBeCalled();
  });
});
