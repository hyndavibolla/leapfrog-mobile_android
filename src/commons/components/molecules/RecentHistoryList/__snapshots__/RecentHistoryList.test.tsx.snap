// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentHistoryList should render 1`] = `null`;
